### PR TITLE
[feat]: [CCM-15455]: Documentation for SMP GCP Support

### PR DESCRIPTION
**Pull request by joeyouss**
# Harness Developer Pull Request

## What Type of PR is This?

- [ ] Issue
- [x] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

File: docs/cloud-cost-management/get-started/ccm-smp/gcp-smp.md [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+**Figure: GCP CCM Self-Managed Enterprise Edition architecture diagram**
+
+<DocImage path={require('./static/gcp-smp-arch.png')} width="50%" height="50%" title="Click to view full size image" />
+
+## Major Components
+- **K8s Job:** In the cloud-hosted infra, a K8s Job is configured to run daily. This job performs all necessary steps to ingest GCP billing data from BigQuery to ClickHouse, utilizing a GCS bucket as an intermediary and then eventually to ClickHouse table.
+- **BigQuery:** The GCP project's billing data resides in a BigQuery table.
+- **Google Cloud Storage (GCS):** Billing data is stored in the parquet format in GCS bucket. Parquet, as a column-oriented format, represents a better interchange format since it is inherently compressed and faster for BigQuery to export and ClickHouse to query.
+- **ClickHouse:** Serves as the datastore where GCP billing data is finally ingested.
+Service Account Key: It is a JSON file that contains the information needed to authenticate and authorize requests made by an application or a service.
```
Comment (sdsingh07) [inline docs/cloud-cost-management/get-started/ccm-smp/gcp-smp.md:21]: Bold is missing

File: docs/cloud-cost-management/get-started/ccm-smp/gcp-smp.md [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+
+
+You need to perform the following tasks to set up CCM for GCP: 
+
+1. [Create Service Account](#create-service-account).
+2. [Create HMAC Key](#Create-HMAC-Key).
+3. [Handling Kubernetes Secrets](#handling-kubernetes-secrets).
+4. [Upload the CF template to S3 bucket](#upload-cf-template-to-s3-bucket).
+5. [Deploy workloads via Helm charts](#deploy-workloads-via-helm-charts).
+
```
Comment (sdsingh07) [inline docs/cloud-cost-management/get-started/ccm-smp/gcp-smp.md:41]: Heading 2: GCP Setup

File: docs/cloud-cost-management/get-started/ccm-smp/gcp-smp.md [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+You need to perform the following tasks to set up CCM for GCP: 
+
+1. [Create Service Account](#create-service-account).
+2. [Create HMAC Key](#Create-HMAC-Key).
+3. [Handling Kubernetes Secrets](#handling-kubernetes-secrets).
+4. [Upload the CF template to S3 bucket](#upload-cf-template-to-s3-bucket).
+5. [Deploy workloads via Helm charts](#deploy-workloads-via-helm-charts).
+   
+## Create Service Account
```
Comment (sdsingh07) [inline docs/cloud-cost-management/get-started/ccm-smp/gcp-smp.md:42]: Heading 3: Step 1: Create Service Account

File: docs/cloud-cost-management/get-started/ccm-smp/gcp-smp.md [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+storage.objects.get
+```
+
+
+2. Follow [these](https://cloud.google.com/iam/docs/service-accounts-create#creating) steps to create Service Account and add the above IAM custom role to it.
+3. Follow [these](https://developers.google.com/workspace/guides/create-credentials#create_credentials_for_a_service_account) steps to create Service Account Credential Key.
+
+Please refer [this](https://www.cloudquery.io/blog/creating-cross-project-service-accounts-in-gcp#how-do-you-set-up-a-service-account-in-gcp?) doc if you encounter any difficulties understanding the process. It provides a comprehensive explanation of all steps accompanied by screenshots.
+
+## Step 2: Create HMAC Key
```
Comment (sdsingh07) [inline docs/cloud-cost-management/get-started/ccm-smp/gcp-smp.md:70]: Heading 3: Step 2: Create HMAC Key

File: docs/cloud-cost-management/get-started/ccm-smp/gcp-smp.md [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+To be able to make use of ClickHouse methods to ingest data from GCS, you will need HMAC key.
+1. Follow [these](https://cloud.google.com/storage/docs/authentication/managing-hmackeys#create) steps to create HMAC key for the Service Account created above.
+
+Deploying workloads via Helm charts
```
Comment (sdsingh07) [inline docs/cloud-cost-management/get-started/ccm-smp/gcp-smp.md:74]: Heading 2: Deploying workloads via Helm charts

File: docs/cloud-cost-management/get-started/ccm-smp/gcp-smp.md [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+Step 1: Clone chart repository
```
Comment (sdsingh07) [inline docs/cloud-cost-management/get-started/ccm-smp/gcp-smp.md:75]: Heading 3

File: docs/cloud-cost-management/get-started/ccm-smp/gcp-smp.md [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+
+```
+git clone git@github.com:harness/helm-charts.git
+cd main/src/harness
+```
+
+Step 2: Already using harness services OnPrem? Upgrade charts
```
Comment (sdsingh07) [inline docs/cloud-cost-management/get-started/ccm-smp/gcp-smp.md:82]: Heading 3

File: docs/cloud-cost-management/get-started/ccm-smp/gcp-smp.md [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+    serviceAccount:
+      create: true
+      annotations: {}
+      name: "batch-processing-default"
+    gcpConfig:
+      GCP_SMP_ENABLED: true
+      bucketNamePrefix: "harness-ccm-%s-%s" # Update the bucket name prefix if you want, but keep double %s for project and region
+```
+
+## Step 3: Handling Kubernetes Secrets
```
Comment (sdsingh07) [inline docs/cloud-cost-management/get-started/ccm-smp/gcp-smp.md:119]: Heading 3

File: docs/cloud-cost-management/get-started/ccm-smp/gcp-smp.md [hunk 9]
```diff
@@ -0,0 +1,201 @@
...
+
+    ```
+    ce-gcp-home-project-creds: <gcpServiceAccountCredentials>
+    ce-batch-gcp-credentials: <gcpServiceAccountCredentials>
+    ```
+    
+3. cloud-info-secret-mount [config-file]
+    ```kubectl edit secret cloud-info-secret-mount -n <namespace>```
+    <details>
+    <summary>config.toml.dist</summary>
```
Comment (sdsingh07) [inline docs/cloud-cost-management/get-started/ccm-smp/gcp-smp.md:144]: We have to add the config-toml.dist file to the doc. Please refer azure PR for the same.

File: docs/cloud-cost-management/get-started/ccm-smp/gcp-smp.md [hunk 10]
```diff
@@ -0,0 +1,201 @@
...
+    SMTP_USERNAME: <SMTP_USERNAME>
+    ```
+
+:::important note
+Note: kubectl edit pvc wal-volume-harness-timescaledb-0 -n <namespace> and increase to 100Gi. It is used by recommendations and anomalies features within CCM services.
+:::
+
+## GCP Connector Setup
+
+## Authorize Service Account to access BigQuery and GCS buckets in other projects
```
Comment (sdsingh07) [inline docs/cloud-cost-management/get-started/ccm-smp/gcp-smp.md:187]: Heading 3

File: docs/cloud-cost-management/get-started/ccm-smp/gcp-smp.md [hunk 11]
```diff
@@ -0,0 +1,201 @@
...
+cd main/src/harness
+```
+
+Step 2: Already using harness services OnPrem? Upgrade charts
+
+```
+helm get values <chart-name> -n <namespace> > override.yaml
+# update override.yaml with ccm specific configuration provided below
+helm upgrade <chart-name> <chart-directory> -n <namespace> -f override.yaml
+# Example: helm upgrade harness . -n harness -f override-prod.yaml
```
Comment (sdsingh07) [inline docs/cloud-cost-management/get-started/ccm-smp/gcp-smp.md:88]: We can remove this Example. Line no. 88

File: docs/cloud-cost-management/get-started/ccm-smp/gcp-smp.md [hunk 12]
```diff
@@ -0,0 +1,201 @@
...
+
+**Steps**
+1. Authenticate and initialize clients for BigQuery, GCS and ClickHouse using SA and HMAC keys.
+2. Create a dedicated GCS bucket to store billing data in Parquet format. The sub-folders will be organized based on timestamps or dates to precisely track when each bucket was created.
+3. Initiate an export query to transfer data from BigQuery to the GCS bucket in Parquet format with GZIP compression. The export query employs a **SELECT** statement, resulting in charges based solely on the data scan.
+4. Check if the billing table exists in **ClickHouse**; create it if not.
+5. Initiate an insert query to efficiently move data from the GCS bucket to the **ClickHouse table** in batch.
+
+
+You need to perform the following tasks to set up CCM for GCP:
```
Comment (sdsingh07) [inline docs/cloud-cost-management/get-started/ccm-smp/gcp-smp.md:34]: 1. [Create Service Account](#create-service-account).
2. [Create HMAC Key](#Create-HMAC-Key).
3. [Deploy workloads via Helm charts](#deploy-workloads-via-helm-charts).
4. [Handling Kubernetes Secrets](#handling-kubernetes-secrets).

File: docs/cloud-cost-management/get-started/ccm-smp/gcp-smp.md [hunk 13]
```diff
@@ -0,0 +1,201 @@
...
+4. Check if the billing table exists in **ClickHouse**; create it if not.
+5. Initiate an insert query to efficiently move data from the GCS bucket to the **ClickHouse table** in batch.
+
+
+You need to perform the following tasks to set up CCM for GCP: 
+
+1. [Create Service Account](#create-service-account).
+2. [Create HMAC Key](#Create-HMAC-Key).
+3. [Handling Kubernetes Secrets](#handling-kubernetes-secrets).
+4. [Upload the CF template to S3 bucket](#upload-cf-template-to-s3-bucket).
```
Comment (sdsingh07) [inline docs/cloud-cost-management/get-started/ccm-smp/gcp-smp.md:39]: What is 4th step?

And, where are they redirecting? We don't have Create-HMAC-Key page. Are they re-directing to heading? If yes, then how the hash definition is mapping with headings?

File: docs/cloud-cost-management/get-started/ccm-smp/gcp-smp.md [hunk 14]
```diff
@@ -0,0 +1,328 @@
...
+1. Authenticate and initialize clients for BigQuery, GCS and ClickHouse using SA and HMAC keys.
+2. Create a dedicated GCS bucket to store billing data in Parquet format. The sub-folders will be organized based on timestamps or dates to precisely track when each bucket was created.
+3. Initiate an export query to transfer data from BigQuery to the GCS bucket in Parquet format with GZIP compression. The export query employs a **SELECT** statement, resulting in charges based solely on the data scan.
+4. Check if the billing table exists in **ClickHouse**; create it if not.
+5. Initiate an insert query to efficiently move data from the GCS bucket to the **ClickHouse table** in batch.
+
+
+You need to perform the following tasks to set up CCM for GCP: 
+
+1. [Create Service Account](#create-service-account).
```
Comment (sdsingh07) [inline docs/cloud-cost-management/get-started/ccm-smp/gcp-smp.md:36]: How these links work? If I click on "Create Service Account", will it redirect me to "Step 1: Create Service Account"?

File: docs/cloud-cost-management/get-started/ccm-smp/gcp-smp.md [hunk 15]
```diff
@@ -0,0 +1,328 @@
...
+    ce-gcp-home-project-creds: <gcpServiceAccountCredentials>
+    ce-batch-gcp-credentials: <gcpServiceAccountCredentials>
+    ```
+    
+3. cloud-info-secret-mount [config-file]
+    ```kubectl edit secret cloud-info-secret-mount -n <namespace>```
+    <details>
+    <summary>config.toml.dist</summary>
+
+    environment = "production"
```
Comment (sdsingh07) [inline docs/cloud-cost-management/get-started/ccm-smp/gcp-smp.md:149]: Add it in code block.

**Review by sdsingh07 — APPROVED** [review]

**Review by rdsoze — APPROVED** [review]

**Pull request merged**